packages/openactive-broker-microservice/app.js

```diff
@@ -534,10 +456,13 @@ async function setFeedIsUpToDate(feedIdentifier) {
 
       // If all feeds are now completed, trigger responses to healthcheck
       if (state.incompleteFeeds.length === 0) {
-        // Stop the validator threads as soon as we've finished harvesting - so only a subset of the results will be validated
-        // Note in some circumstances threads will complete their work before terminating
-        await Promise.all(state.validatorThreadArray.map(async (validator) => validator.terminate));
-        // TODO surely this doesn't work ^ needs to actually call terminate()
+        /* Signal for the Validator Worker Pool that we may stop once the validator timeout has run.
+        Validator is an expensive process and is not completely necessary for Booking API testing. So we put a hard
+        limit on how long it runs for (once all items are harvested).
+        This means that, in some cases, only a subset of the results will be validated.
+        Note that the worker pool will finish its current iteration if it has already reached the timeout. */
+        await validatorWorkerPool.stopWhenTimedOut();
```

Comment:
This is great! Just noting that when in `VALIDATE_ONLY` mode it should instead wait for validation to complete across all items (rather than killing validation early when harvesting is complete)

This might be one for a separate task, but thought I'd flag here in case it was useful to note how this might be achieved while you're in the context?

Reply:
Thanks @nickevansuk! Silly me, hadn't realised that.

Have fixed this now. If VALIDATE_ONLY=true, then WAIT_FOR_HARVEST=false, in which case validatorTimeout=null, in which case, **ValidatorWorkerPool will not wait for a timeout**

Most of that logic was already in place, but the bold bit wasn't.

Looking at the code though, it seems like this might have been broken before? As, if `validatorTimeoutMs=null`, then in that previous logic, the condition at [async-validator-worker.js:19](https://github.com/openactive/openactive-test-suite/pull/479/files#diff-017ca6e804ec1ceb0b34fc727657feab462cc9055c089ccc172fe2dfb80ca2e1L19), `remainingValidationTimeoutMs > 0` will always be true because `<number> - null` = `<number>` in JS. So assuming that my understanding of what's supposed to happen here is correct, this looks to be fixed!